integration-tests/tests/test_identity_transformation.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+
+import pytest
+from tests.utils.config import (
+    DatasetLogs,
+    PackageConfig,
+)
+from tests.utils.utils import (
+    is_dir_tree_content_equal,
+    is_json_file_structurally_equal,
+    run_and_assert,
+)
+
+pytestmark = pytest.mark.binaries
+
+text_datasets = pytest.mark.parametrize(
+    "dataset_logs_fixture",
+    [
+        "hive_24hr",
+    ],
+)
+
+json_datasets = pytest.mark.parametrize(
+    "dataset_logs_fixture",
+    [
+        "elasticsearch",
+        "spark_event_logs",
+        "postgresql",
+    ],
+)
+
+
+@pytest.mark.clp
+@text_datasets
+def test_clp_identity_transform(
+    request,
+    package_config: PackageConfig,
+    dataset_logs_fixture: str,
+) -> None:
+    dataset_logs: DatasetLogs = request.getfixturevalue(dataset_logs_fixture)
+    dataset_name = dataset_logs.name
+    dataset_logs.clear_test_outputs()
+
+    binary_path_str = str(package_config.clp_bin_dir / "clp")
+    # fmt: off
+    compression_cmd = [
+        binary_path_str,
+        "c",
+        "--progress",
+        "--remove-path-prefix", str(dataset_logs.extraction_dir),
+        str(dataset_logs.compression_dir),
+        str(dataset_logs.extraction_dir),
+    ]
+    # fmt: on
+    run_and_assert(compression_cmd)
+
+    extraction_cmd = [
+        binary_path_str,
+        "x",
+        str(dataset_logs.compression_dir),
+        str(dataset_logs.decompression_dir),
+    ]
+    run_and_assert(extraction_cmd)
+
+    assert is_dir_tree_content_equal(
+        dataset_logs.extraction_dir, dataset_logs.decompression_dir
+    ), "Mismatch between clp compression input and decompression output."
+
+    dataset_logs.clear_test_outputs()
+
+
+@pytest.mark.clp_s
+@json_datasets
+def test_clp_s_identity_transform(
+    request,
+    package_config: PackageConfig,
+    dataset_logs_fixture: str,
+) -> None:
+    dataset_logs: DaatsetLogs = request.getfixturevalue(dataset_logs_fixture)
+    dataset_name = dataset_logs.dataset_name
+    _clp_s_compress_and_decompress(package_config, dataset_logs)
+
+    # Recompress the decompressed output that's consolidated into a single json file, and decompress
+    # it again to verify consistency. The compression input of the second iteration points to the
+    # decompression output of the first.
+    # TODO: Remove this check once we can directly compare decompressed logs (which would preserve
+    #       the directory structure and row/key order) with the original downloaded logs.
+    # See also: https://docs.yscope.com/clp/main/user-guide/core-clp-s.html#current-limitations
+    consolidated_json_logs = DatasetLogs.create(
+        package_config=package_config,
+        dataset_name=f"{dataset_name}-consolidated-json",
+        extraction_dir=dataset_logs.decompression_dir,
+    )
```

Comment:
To be honest, I don't like the new design of overloading the DatasetLogs. Imo, it would be simpler to only let DatasetLogs manage downloaded log. Creating a "pseudo" dataset at the middle of the test doesn't feel intuitive, and it also make the DatasetLogs interface more complicated as it now needs to handle a list of optional arguments.

If the whole purpose is for reuse in _clp_s_compress_and_decompress, I think we can extend the argument list of _clp_s_compress_and_decompress instead, or we introduce something like test_config.

That said, this is just my opinion, if you don't agree, we can discuss with Kirk